tcp/src/dispatch.rs

```diff
@@ -3,24 +3,25 @@ use std::collections::HashMap;
 use std::net::SocketAddr;
 use std::sync::Arc;
 
-use crate::jsonrpc::futures::sync::mpsc;
-use crate::jsonrpc::futures::{Async, Future, Poll, Sink, Stream};
+use crate::jsonrpc::futures::{self as futures03, channel::mpsc, StreamExt};
```

Comment:
I'd prefer to use the re-exported one to avoid rewriting the code in the future. If `tcp` get's rid of compat requirements we can simply remove the `futures03` dependency from Cargo.toml. Added a todo for this.